lib/Internal/Vhost.php

```diff
@@ -29,37 +31,22 @@ class Vhost implements Monitor {
         "crypto_method"         => STREAM_CRYPTO_METHOD_SSLv23_SERVER, /* means: TLS 1.0 and up */
     ];
 
-    private static $cryptoMethodMap = [
-        "tls"       => STREAM_CRYPTO_METHOD_SSLv23_SERVER, // no, not STREAM_CRYPTO_METHOD_TLS_SERVER, which is equivalent to TLS v1.0 only
-        "tls1"      => STREAM_CRYPTO_METHOD_TLSv1_0_SERVER,
-        "tlsv1"     => STREAM_CRYPTO_METHOD_TLSv1_0_SERVER,
-        "tlsv1.0"   => STREAM_CRYPTO_METHOD_TLSv1_0_SERVER,
-        "tls1.1"    => STREAM_CRYPTO_METHOD_TLSv1_1_SERVER,
-        "tlsv1.1"   => STREAM_CRYPTO_METHOD_TLSv1_1_SERVER,
-        "tls1.2"    => STREAM_CRYPTO_METHOD_TLSv1_2_SERVER,
-        "tlsv1.2"   => STREAM_CRYPTO_METHOD_TLSv1_2_SERVER,
-        "ssl2"      => STREAM_CRYPTO_METHOD_SSLv2_SERVER,
-        "sslv2"     => STREAM_CRYPTO_METHOD_SSLv2_SERVER,
-        "ssl3"      => STREAM_CRYPTO_METHOD_SSLv3_SERVER,
-        "sslv3"     => STREAM_CRYPTO_METHOD_SSLv3_SERVER,
-        "any"       => STREAM_CRYPTO_METHOD_ANY_SERVER,
-    ];
-
     /** @Note Vhosts do not allow wildcards, only separate 0.0.0.0 and :: */
-    public function __construct(string $name, array $interfaces, callable $application, array $filters, array $monitors = [], HttpDriver $driver = null) {
+    public function __construct(string $name, array $interfaces, Responder $responder, array $middlewares = [], array $monitors = []) {
```

Comment:
Oh. Why did you remove the possibility to specify an a specific/own http driver? (e.g. you want to be HTTP/2 only, or you want to implement a proxy http driver etc.)

Reply:
Nobody ever used that. If you just want HTTP/2, that's better solved with an option. It's an internal detail that can change at any time if it's internal. We don't want to provide a BC guarantee here.